sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGenerateExec.scala

```diff
@@ -164,6 +180,61 @@ trait GpuGenerator extends GpuUnevaluable {
   }
 }
 
+case class GpuReplicateRows(children: Seq[Expression]) extends GpuGenerator with ShimExpression {
+
+  override def elementSchema: StructType =
+    StructType(children.tail.zipWithIndex.map {
+      case (e, index) => StructField(s"col$index", e.dataType)
+    })
+
+  override def generate(inputBatch: ColumnarBatch,
+      generatorOffset: Int,
+      outer: Boolean): ColumnarBatch = {
+
+    val schema = GpuColumnVector.extractTypes(inputBatch)
+    val replicateVector = GpuColumnVector.extractBases(inputBatch)(generatorOffset)
```

Comment:
nit: I was thinking more of something like

```
val replicateVector = inputBatch.column(generatorOffset).asInstanceOf[GpuColumnVector]
```

Or if we are converting `inputBatch` to a table anyways...

```
    withResource(GpuColumnVector.from(inputBatch)) { table =>
      val replicateVector = table.getColumn(generatorOffset)
      withResource(table.repeat(replicateVector)) { replicatedTable =>
        GpuColumnVector.from(replicatedTable, schema)
      }
    }
```